components/camel-spring-ldap/src/main/java/org/apache/camel/component/springldap/SpringLdapProducer.java

```diff
@@ -89,12 +91,19 @@ public void process(Exchange exchange) throws Exception {
             throw new UnsupportedOperationException("LDAP operation must not be empty, but you provided an empty operation");
         }
 
+        LdapTemplate ldapTemplate = endpoint.getLdapTemplate();
+
         String dn = (String)body.get(DN);
+        if (StringUtils.isBlank(dn)) {
+            ContextSource contextSource = ldapTemplate.getContextSource();

```

Comment:
it was just an opinion. despite of having such setting in context and using the dn coming from exchange makes having dn set in context useless. it did not make me feel right. i feel like this is an implementation choice, that's why i proposed an option on the endpoint whether to choose from context or camel exchange. 

https://issues.apache.org/jira/browse/CAMEL-11060?focusedCommentId=15955226&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15955226